### PR TITLE
Enable auto sync per default for Open/Libre Office

### LINKED ISSUE #6957: Enable Auto Sync per default for Open/LibeOffice
JabRef 5.1--2020-08-30--e023aa0
Ubuntu 18.04.3 LTS 
Java 14.0.2
Libreoffice 6.0.7.3

- [x] **Mandatory**: I have tested the latest development version from http://builds.jabref.org/master/ and the problem persists

Instead of showing a Number in brackets as citation when using the "default [Number] style file" a minus sign with space "- " is printed when citing in LibreOffice. The "default [Author] style file" works as expected though.

1. Connect to LibreOffice Document
2. Select citation style "Default [Number] style file"
3. Cite a reference by clicking on "cite" or "cite in text"
4. The citation in Libreoffice will only contain a "- " and no Number or brackets

The bug can be reproduced with any reference I choose. I also tried to define my own copy of the default number style file from this file src/main/resources/resource/openoffice/default_numerical.jstyle but I get the same output (minus sign with space instead of Number in brackets).

### PR DESCRIPTION
**Pull request by IsaacRoles**
Enabling Automatically sync bibliography when inserting citations as the default for Open/Libre Office integrations
Fixes #6957 

Pic 1

I have also made changes to the OpenOffice/LibreOffice integration documentation and will submit a pull request for those changes.

- [x] Change in CHANGELOG.md described (if applicable)
- [x] Tests created for changes (if applicable)
- [x] Manually tested changed features in running JabRef (always required)
- [x] Screenshots added in PR description (for UI changes)
- [x] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, submitted a pull request to the documentation repository.

### CODE REVIEW

**Comment by IsaacRoles** [comment]
I will wait for feedback before submitting a pull request for the documentation changes.

File: src/test/java/org/jabref/preferences/OpenOfficeSyncWhenCitingTest.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+
+public class OpenOfficeSyncWhenCitingTest {
+
+    private static boolean previousValue;
+    private final JabRefPreferences preferences = JabRefPreferences.getInstance();
```
Comment (Siedlerchr) [inline src/test/java/org/jabref/preferences/OpenOfficeSyncWhenCitingTest.java:12]: Thanks for thinking of a test, but you don't need one here as you are just testing that a variable is stored in the preferences which is not that critical.  In JabRef we focus on testing the important things, e.g. model/logic.

To sum up => Just remove the test and it's all good.

And there is a general side effect you have here with the preferences in test , by using the JabRefPreferences Instance Object you are modifying the real preferences values which are also stored/persisted in the registry etc.  Therefore, if you need to test something which depends on the preferencesm you need to mock them and "fake" the return values. 
See for example:
```
   importFormatPreferences = mock(ImportFormatPreferences.class, Answers.RETURNS_DEEP_STUBS);
        when(importFormatPreferences.getKeywordSeparator()).thenReturn(',');
```

**Review by Siedlerchr — CHANGES_REQUESTED** [review]
Thanks for your contribution! As I explained above, you can just remove the test.

File: src/test/java/org/jabref/preferences/OpenOfficeSyncWhenCitingTest.java (hunk 1 above)
Comment (IsaacRoles) [reply src/test/java/org/jabref/preferences/OpenOfficeSyncWhenCitingTest.java]: Thanks for your feedback! I went ahead and removed that test. Thanks for your advice, I wasn't sure what would be worth testing.

**Review by koppor — APPROVED** [review]
For me, it looks good. Not tested though.

**Comment by Siedlerchr** [comment]
Thanks for your contribution!

**Comment by teertinker** [comment]
Has this feature already been tested for typical use cases? I hope that there are no drawbacks. If the library is synced every time I see two possible problems.
1. Parsing long manuscripts takes sometimes a lot of time, if the list of references is refreshed each and every time.
2. The workflow in libreoffice may get disrupted. JabRef pushes many commands to libreoffice while refreshing. As a result, users probably cannot use the edit/undo function anymore.

Since users have the choice, one could still deactivate automatic sync. The problem, however, is that many users do not know that the function exists and what implications it has. I think this is also the reason for issue #6957 

I would opt for leaving automatic sync off, because this matches the current design better (we have a refresh button).

**Comment by IsaacRoles** [comment]
I can definitely see your point @teertinker. I don't have enough experience with JabRef to know for sure what the best option is. I can see how this may solve an issue for one particular user (in the case of issue #6957), but cause workflow issues for other users. I needs to be determined what the most common use case is and what is most intuitive to the user.

**Comment by koppor** [comment]
@teertinker Maybe, this is a discussion between more "modern" users? I have the impression that many things happen very automatically these days - and the users are more asked to fix somethign if something goes wrong. For instance, IntelliJ just saves the file upon change. Overleaf just compiles the document during writing, ... - I would also wait for feedback. Maybe, there are not that many users crafting long documents in that setting (comapred to users just having 10 to 20 references, such as students). I would also guess/hope/pray that advanced users know about the setting.

**Comment by teertinker** [comment]
I did a short test using an article with about 20 pages and about 40 references, which should be a typical user case. To test, I inserted a new citation and switched to libreoffice afterwards

performance
-----------------
- My new laptop (intel I5) did the sync pretty fast. After switching to libreoffice I can see that there is some text modification going on, but the workflow is not affected.
- On my old laptop (Intel i3 but 7 years old) I had to wait until sync is completed.

I think the overall **performance** for the average user should be ok.

workflow
--------------
One problem already stated above is that you can't use the undo/redo commands in Libreoffice after inserting a citation because this would only undo/redo the recent internal modifications made by libreoffice. If a user is writing a sentence, cites something and than want's to revert changes by using CTRL+Z this won't work.

I think this is no critical issue but the **workflow** will be disrupted in some cases without the user knowing exactly why this happens.

automatic sync and sync is not the same
--------------
My manuscript contained a reference, that was not included in my current bib file. When I click on the sync button, JabRef will not sync the library but shows an error message that entry xy does not exist.  When I switch automatic sync on, however, the library gets synced - except for the missing entry. No error message appears.
 
While this feature of automatic sync is really good for the workflow and should be kept, it might be an issue for typical user cases. I was assuming, that automatic sync and manual sync have the same functionality and thus would have missed that one reference in my manuscript is corrupted and hence get's not cited properly.

summary
-------------
Different users have different needs. I completely understand the advantage of automatic sync when using numbered citations. In my case, I definitely will switch it off - and thanks to the philosophy of JabRef I can. When writing a manuscript I insert citations and do not care about the style of my references. The style (and hence the sync function) is only needed as a last step.

**Pull request merged**